docs/introduction.rst

```diff
@@ -107,17 +109,49 @@ structure could be like a simplified XML DOM-tree with following specification:
   (e.g. the unit of the data in the node, etc.). To make things simple, the
   attributes should be text attributes, like in XML.
 
-The sender would assemble a tree with the necessary information and transmit it via
-the transport layer. The receiver would then query the received tree, look
+The sender would assemble a tree with the necessary information and transmit it
+via the transport layer. The receiver would then query the received tree, look
 for the presence / absence of given nodes and extract the necessary information
 from the tree.
 
 I have already started a small C-library with this functionality, the `saydx
-library <https://github.com/saydx/saydx>`_. Although not finished yet, it could
-be used as the message layer. It would provide the basic infrastructure for tree
-manipulation, as well as routines to read and write trees to file or to pass
-them from C to Fortran and vice-versa. Combined with cslib, it could cover the
-functionality of the first two layers.
+[səeɪdɪks] library <https://github.com/saydx/libsaydx>`_. Although not finished
+yet, it could be used as the message layer. It would provide the basic
+infrastructure for tree manipulation, as well as routines to read and write
+trees to file or to pass them from C to Fortran and vice-versa. Combined with
+cslib, it could cover the functionality of the first two layers.
+
+
+Array types
+^^^^^^^^^^^
+
+The message layer should understand / support following array data types:
+
+* real numbers (4, 8 and eventually 16 bytes)
+* complex numbers (composed of two real numbers)
+* integers (4 and 8 bytes)
+* logicals (represented by integers)
+* characters (1 byte)
+* strings (of arbitrary length)
+* bytes
+
+When the data is represented in binary form, the native representation of the
+x86_64 architecture should be used. So, at least in its first version, the
+binary version of the protocol won't be architecture independent. We could allow
+for passing the arrays also in text form if this ever becomes an issue. (The
+saydx library can already store the tree in text form.) Implementing
+architecture independence on the binary level (as in HDF5) would be probably an
+overkill.
```

Comment:
Not sure, whether this can be realized so easily. It would definitely mean that the user has to provide C-functions (even if she uses the Fortran interface) and change / extend some core parts of the library. As a workaround, the user could convert his array to a binary blob and send that as data...

Reply:
Not sure if it has to be C. If its via a text string, the binding would be transparent to the language on top. At the simplest level, pointing out that rendering things in/out of a string on the user side give this functionality.